### DISCUSSION TITLE
e2etests: Make e2e-Tests standalone

### DISCUSSION THREAD

**Discussion by Stezido** [openkfw/TruBudget#874] [General]
Currently the e2e tests are tightly coupled with provisioned data and some values (e.g. root-secret) are hardcoded. The e2e-tests should be running without any provisioning and should be "rerun"- able.

### Tasks todo:
- [ ] Decouple tests ( Closes #151 )
- [ ] Outsource hardcoded variables

**Comment by daniel-arnauer** [comment]
There are 2 different provisiongs for TEST and PROD:
- TruBudget\provisioning\src\data\prod
- TruBudget\provisioning\src\data\test

The e2e-tests are currently associated with the test provisioning. This is why an error occurs when setting the Environment to PROD (#151).

For me, it makes no sense to make all tests standalone because we need pre-set data in the UI. 
If there is an cypress error because it cannot find an element if a list is too long, the test should be written more explicit, e.g. use .scollIntoView() method.

**Comment by Stezido** [comment]
Why we need pre-set data? The e2e-tests could generate this data and ensure the data is there before running each test.

**Comment by daniel-arnauer** [comment]
That is an option, but one disadvantage is the speed, the e2e-tests will need more time if it has to create the data itself. 
And the problem with too big user lists for cypress (scrollIntoView) will not be resolved this way
